src/coreclr/src/jit/emit.cpp

```diff
@@ -5880,19 +5942,33 @@ void emitter::emitOutputDataSec(dataSecDsc* sec, BYTE* dst)
 #ifdef DEBUG
             if (EMITVERBOSE)
             {
-                printf("  ");
+                printf("  section %3u, size %2u, RWD%2u:\t", secNum++, dscSize, curOffs);
+
                 for (size_t i = 0; i < dscSize; i++)
                 {
                     printf("%02x ", dsc->dsCont[i]);
                     if ((((i + 1) % 16) == 0) && (i + 1 != dscSize))
                     {
-                        printf("\n  ");
+                        printf("\n\t\t\t\t\t");
                     }
                 }
+                switch (dsc->dsDataType)
+                {
+                    case TYP_FLOAT:
+                        printf(" ; float  %9.6g", (double)*reinterpret_cast<float*>(&dsc->dsCont));
```

Comment:
Why is this converting to `double` for `printf` rather than just keeping it as `float`?

Reply:
AFAIK printf() only supports the printing of doubles

Reply:
The C  calling convention only passes doubles, not floats

Reply:
👍, I'm definitely not as up to date with what the official standards support.

A simple MSVC console app, however, looks to work and Clang/GCC don't give any warnings on compile.